lib/active_merchant/billing/gateways/safe_charge.rb

```diff
@@ -152,7 +152,7 @@ def add_payment(post, payment, options = {})
 
         if payment.is_a?(NetworkTokenizationCreditCard) && payment.source == :network_token
           post[:sg_CAVV] = payment.payment_cryptogram
-          post[:sg_ECI] = options[:three_d_secure][:eci] || '05'
+          post[:sg_ECI] = options[:three_d_secure] && options[:three_d_secure][:eci] ? options[:three_d_secure][:eci] : '05'
```

Comment:
Could this be simplified to the following?

post[:sg_ECI] = `options[:three_d_secure] && options[:three_d_secure][:eci] || '05'`

Reply:
yep, done.